In [1]:
import os
import pandas as pd

## One time preprocessing work

In [2]:
# Assuming working directory is in folder where all of the json data is located 
# os.getcwd()

In [3]:
# If yelp data is downloaded as a tar file, the data can be extracted with the following code

# my_tar = tarfile.open('my_tar.tar.gz')
# my_tar.extractall('./my_folder') # specify which folder to extract to
# my_tar.close()

In [4]:
# Converting business json file to a csv for ease of use (one time only)
#bus = pd.read_json('business.json', lines = True)
#bus.to_csv(r"business.csv", index = False)

In [5]:
# Converting review json file to a csv for ease of use (one time only)
#rev = pd.read_json('review.json', lines = True)
#rev.to_csv('review.csv', index = False)

## Checking yelp data

In [6]:
# Checking the business.csv dataframe
business = pd.read_csv('business.csv')
print('Entire business data set shape:', business.shape)
business.head()


Entire business data set shape: (192609, 14)


address  \
0     2818 E Camino Acequia Drive   
1            30 Eglinton Avenue W   
2       10110 Johnston Rd, Ste 15   
3   15655 W Roosevelt St, Ste 237   
4  4209 Stuart Andrew Blvd, Ste F   

                                          attributes             business_id  \
0                           {'GoodForKids': 'False'}  1SWheh84yJXfytovILXOAQ   
1  {'RestaurantsReservations': 'True', 'GoodForMe...  QXAEGFB4oINsVuTFxEYKFQ   
2  {'GoodForKids': 'True', 'NoiseLevel': "u'avera...  gnKjwL_1w79qoiV3IC_xQQ   
3                                                NaN  xvX2CttrVhyG2z1dFg_0xw   
4  {'BusinessAcceptsBitcoin': 'False', 'ByAppoint...  HhyxOkGAM07SRYtlQ4wMFQ   

                                          categories         city  \
0                                  Golf, Active Life      Phoenix   
1  Specialty Food, Restaurants, Dim Sum, Imported...  Mississauga   
2                  Sushi Bars, Restaurants, Japanese    Charlotte   
3                      Insurance, Financial Services     Goodyear   
4  Plumbing, Shopping, Local Services, Home Servi...    Charlotte   

                                               hours  is_open   latitude  \
0                                                NaN        0  33.522143   
1  {'Monday': '9:0-0:0', 'Tuesday': '9:0-0:0', 'W...        1  43.605499   
2  {'Monday': '17:30-21:30', 'Wednesday': '17:30-...        1  35.092564   
3  {'Monday': '8:0-17:0', 'Tuesday': '8:0-17:0', ...        1  33.455613   
4  {'Monday': '7:0-23:0', 'Tuesday': '7:0-23:0', ...        1  35.190012   

    longitude                             name postal_code  review_count  \
0 -112.018481       Arizona Biltmore Golf Club       85016             5   
1  -79.652289       Emerald Chinese Restaurant     L5R 3E7           128   
2  -80.859132      Musashi Japanese Restaurant       28210           170   
3 -112.395596  Farmers Insurance - Paul Lorenz       85338             3   
4  -80.887223              Queen City Plumbing       28217             4   

   stars state  
0    3.0    AZ  
1    2.5    ON  
2    4.0    NC  
3    5.0    AZ  
4    4.0    NC

In [7]:
# Since the scope of the project is for cities near the Las Vegas area, We will subset the data for businesses in Nevada

#******* Should we clean the data to include misspellings and close associations??? ************
business_NV = business[business['state'] == 'NV']
# (36312, 14)
# print('State of Nevada subset shape: ', business_NV.shape)
business_NV['city'].value_counts()

## Subsetting only Las Vegas
# (29361, 14)

business_LV = business_NV[business_NV['city'] == 'Las Vegas']
print('City of Las Vegas subset shape: ', business_LV.shape)
business_LV.head(5)

City of Las Vegas subset shape:  (29361, 14)


address  \
7   4545 E Tropicana Rd Ste 8, Tropicana   
17          1775 E Tropicana Ave, Ste 29   
18                                   NaN   
20               450 Fremont St, Ste 179   
22                   6007 Dean Martin Dr   

                                           attributes             business_id  \
7   {'RestaurantsPriceRange2': '3', 'GoodForKids':...  gbQN7vr_caG_A1ugSmGhWg   
17  {'OutdoorSeating': 'False', 'BusinessAcceptsCr...  PZ-LZzSlhSe9utkQYU8pFg   
18             {'BusinessAcceptsCreditCards': 'True'}  nh_kQ16QAoXWwqZ05MPfBQ   
20                            {'GoodForKids': 'True'}  dFMxzHygTy6F873843dHAA   
22  {'BikeParking': 'False', 'Ambience': "{'romant...  KWywu2tTEPWmR9JnBc0WyQ   

                                           categories       city  \
7   Hair Salons, Hair Stylists, Barbers, Men's Hai...  Las Vegas   
17                               Restaurants, Italian  Las Vegas   
18  Event Planning & Services, Photographers, Prof...  Las Vegas   
20                      Arcades, Arts & Entertainment  Las Vegas   
22  Nightlife, Arts & Entertainment, Bars, Strip C...  Las Vegas   

                                                hours  is_open   latitude  \
7   {'Monday': '10:0-19:0', 'Tuesday': '10:0-19:0'...        1  36.099872   
17                                                NaN        0  36.100016   
18  {'Monday': '0:0-0:0', 'Tuesday': '0:0-0:0', 'W...        1  36.116549   
20  {'Monday': '11:0-0:0', 'Tuesday': '11:0-0:0', ...        1  36.169993   
22  {'Thursday': '19:30-2:0', 'Friday': '19:30-3:0...        1  36.080168   

     longitude                        name postal_code  review_count  stars  \
7  -115.074574                   Supercuts       89121             3    3.5   
17 -115.128528  Carluccio's Tivoli Gardens       89119            40    4.0   
18 -115.088115    Myron Hensel Photography       89121            21    5.0   
20 -115.140685              Fremont Arcade       89101            38    4.5   
22 -115.182756                Hunk Mansion       89118           107    4.0   

   state  
7     NV  
17    NV  
18    NV  
20    NV  
22    NV

In [8]:
# Checking the review.csv dataframe
review = pd.read_csv('review.csv')
print('The shape of the yelp review is: ', review.shape)
review.head()

The shape of the yelp review is:  (6685902, 9)


business_id  cool                 date  funny  \
0  ujmEBvifdJM6h6RLv4wQIg     0  2013-05-07 04:34:36    1.0   
1  NZnhc2sEQy3RmzKTZnqtwQ     0  2017-01-14 21:30:33    0.0   
2  WTqjgwHlXbSFevF32_DJVw     0  2016-11-09 20:09:03    0.0   
3  ikCg8xy5JIg_NGPx-MSIDA     0  2018-01-09 20:56:38    0.0   
4  b1b1eb3uo-w561D0ZfCEiQ     0  2018-01-30 23:07:38    0.0   

                review_id  stars  \
0  Q1sbwvVQXV2734tPgoKj4Q    1.0   
1  GJXCdrto3ASJOqKeVWPi6Q    5.0   
2  2TzJjDVDEuAW6MR5Vuc1ug    5.0   
3  yi0R0Ugj_xUx_Nek0-_Qig    5.0   
4  11a8sVPMUFtaC7_ABRkmtw    1.0   

                                                text  useful  \
0  Total bill for this horrible service? Over $8G...     6.0   
1  I *adore* Travis at the Hard Rock's new Kelly ...     0.0   
2  I have to say that this office really has it t...     3.0   
3  Went in for a lunch. Steak sandwich was delici...     0.0   
4  Today was my second out of three sessions I ha...     7.0   

                  user_id  
0  hG7b0MtEbXx5QzbzE6C_VA  
1  yXQM5uF2jS6es16SJzNHfg  
2  n6-Gk65cPZL6Uz8qRm3NYw  
3  dacAIZ6fTM6mqwW5uxkskg  
4  ssoyf2_x0EQMed6fgHeMyQ

In [9]:
# lol
review.loc[0, 'text']

'Total bill for this horrible service? Over $8Gs. These crooks actually had the nerve to charge us $69 for 3 pills. I checked online the pills can be had for 19 cents EACH! Avoid Hospital ERs at all costs.'

## Checking inspection data

In [10]:
restaurant_categories = pd.read_csv('restaurant_categories.csv', delimiter = ';')
restaurant_categories.head()

category_id    category_name  sort_order active  Unnamed: 4
0            2     Bakery Sales           9      N         NaN
1            3  Banquet Kitchen           9      N         NaN
2            4  Banquet Support           9      N         NaN
3            5     Bar / Tavern           2      Y         NaN
4            6         Barbeque           3      Y         NaN

In [11]:
# Extra column that needs to be dropped
restaurant_categories = restaurant_categories.drop('Unnamed: 4', axis = 1)

In [12]:
# Checking to ensure the column has been dropped
restaurant_categories.head(5)

category_id    category_name  sort_order active
0            2     Bakery Sales           9      N
1            3  Banquet Kitchen           9      N
2            4  Banquet Support           9      N
3            5     Bar / Tavern           2      Y
4            6         Barbeque           3      Y

In [13]:
restaurant_establishments = pd.read_csv('restaurant_establishments.csv', delimiter = ';')
restaurant_establishments.head(10)

# ********Check 6 - 9******

permit_number facility_id  owner_id    PE              restaurant_name  \
0     PR0000002   FA0002677       NaN  1006     McDonald's #3549 D HOTEL   
1     PR0000003   FA0002678       NaN  1008           MCDONALD'S  #19724   
2     PR0000004   FA0002679       NaN  1007             McDonald's #6965   
3     PR0000005   FA0002680       NaN  1007             MCDONALD'S #2922   
4     PR0000006   FA0002681       NaN  1006     MCDONALD'S #3153 @ PLAZA   
5     PR0000007   FA0002682       NaN  1008  McDonald's Restaurant 31454   
6     PR0000009   FA0002684       NaN  1016   Larry's Villa Bar & Lounge   
7     PR0000010   FA0002684       NaN  1009      Larry's Villa Snack Bar   
8     PR0000011   FA0002685       NaN  1016            PEPPERMILL LOUNGE   
9     PR0000012   FA0002685       NaN  1003        PEPPERMILL RESTAURANT   

                       location_name                        address  \
0  McDonalds (Store #3549) @ D HOTEL                 301 Fremont St   
1              McDonalds Corporation         4400 E Charleston Blvd   
2                    McDonalds #6965               3175 N Rancho Dr   
3            McDonalds (Store #2922)            2020 N Rainbow Blvd   
4    McDonalds (Store #3153) @ PLAZA                    1 S Main St   
5       McDonald's Restaurant #31454  10590 Southern Highlands Pkwy   
6             Larrys Villa - DELETED              2401 W Bonanza Rd   
7             Larrys Villa - DELETED              2401 W Bonanza Rd   
8                         PEPPERMILL          2985 S Las Vegas Blvd   
9                         PEPPERMILL          2985 S Las Vegas Blvd   

    latitude   longitude  city_id  ... nciaa plan_review record_status  \
0  36.169390 -115.143084       10  ...     Y         NaN             1   
1  36.159439 -115.079751       10  ...     Y         NaN             1   
2  36.216746 -115.212421       10  ...     Y         NaN             1   
3  36.196385 -115.240740       10  ...     Y         NaN             1   
4  36.172176 -115.146829       10  ...     Y         NaN             1   
5  35.997076 -115.207542       10  ...     Y         NaN             1   
6  36.176648 -115.174090       10  ...     N         NaN             2   
7  36.176648 -115.174090       10  ...     N         NaN             2   
8  36.133616 -115.162998       10  ...     Y         NaN             1   
9  36.133616 -115.162998       10  ...     Y         NaN             1   

   current_grade  current_demerits         date_current  previous_grade  \
0              A               6.0  2020-03-13 00:00:00               A   
1              A               3.0  2020-02-06 00:00:00               A   
2              A               0.0  2020-02-10 00:00:00               A   
3              A               0.0  2019-12-10 00:00:00               A   
4              A               0.0  2020-03-10 00:00:00               A   
5              A               6.0  2019-04-25 00:00:00               A   
6              A               3.0  2016-04-14 00:00:00               P   
7              A               3.0  2016-04-14 00:00:00               P   
8              A               0.0  2020-02-04 00:00:00               A   
9              A               8.0  2020-01-16 00:00:00               A   

         date_previous                                        search_text  \
0  2020-03-13 00:00:00    McDonald's #3549 D HOTEL McDonalds 3549 D HOTEL   
1  2020-02-06 00:00:00                MCDONALD'S  #19724 MCDONALDS  19724   
2  2020-02-10 00:00:00                    McDonald's #6965 McDonalds 6965   
3  2019-12-10 00:00:00                    MCDONALD'S #2922 MCDONALDS 2922   
4  2020-03-10 00:00:00     MCDONALD'S #3153 @ PLAZA MCDONALDS 3153  PLAZA   
5  2019-04-25 00:00:00  McDonald's Restaurant 31454 McDonalds Restaura...   
6  2015-10-01 00:00:00  Larry's Villa Bar & Lounge Larrys Villa Bar  L...   
7  2015-10-01 00:00:00     Larry's Villa Snack Bar Larrys Villa Snack Bar   
8  2020-02-04 00:00:00                PEPPERMILL 

In [14]:
restaurant_establishments.columns

Index(['permit_number', 'facility_id', 'owner_id', 'PE', 'restaurant_name',
       'location_name', 'address', 'latitude', 'longitude', 'city_id',
       'city_name', 'zip_code', 'nciaa', 'plan_review', 'record_status',
       'current_grade', 'current_demerits', 'date_current', 'previous_grade',
       'date_previous', 'search_text', 'Unnamed: 21'],
      dtype='object')

In [15]:
# Extra column that needs to be dropped
restaurant_establishments = restaurant_establishments.drop('Unnamed: 21', axis = 1)

In [16]:
restaurant_categories.head()

category_id    category_name  sort_order active
0            2     Bakery Sales           9      N
1            3  Banquet Kitchen           9      N
2            4  Banquet Support           9      N
3            5     Bar / Tavern           2      Y
4            6         Barbeque           3      Y

In [17]:
restaurant_inspection_types = pd.read_csv('restaurant_inspection_types.csv', delimiter = ';')

In [18]:
# Extra column that needs to be dropped
restaurant_inspection_types = restaurant_inspection_types.drop('Unnamed: 2', axis = 1)
restaurant_inspection_types

inspection_type_id                inspection_type
0                 604  Epidemiological Investigation
1                 902        Complaint Investigation
2                 914                  Re-inspection
3                 916             Routine Inspection
4                 918     Special Request Inspection
5                 919                         Survey
6                 928       Final Remodel Inspection

In [19]:
restaurant_cities = pd.read_csv('restaurant_cities.csv', delimiter = ';')
print(restaurant_cities.shape)
restaurant_cities.head()

(26, 3)


city_id        city_name  Unnamed: 2
0        1     Blue Diamond         NaN
1        2     Boulder City         NaN
2        3      Cal-Nev-Ari         NaN
3        4  Cottonwood Cove         NaN
4        5         Glendale         NaN

In [20]:
restaurant_cities = restaurant_cities.drop('Unnamed: 2', axis = 1)
restaurant_cities.head()

city_id        city_name
0        1     Blue Diamond
1        2     Boulder City
2        3      Cal-Nev-Ari
3        4  Cottonwood Cove
4        5         Glendale

In [21]:
restaurant_inspections = pd.read_csv('restaurant_inspections.csv', delimiter = ';')

restaurant_inspections.head()

serial_number permit_number facility_id  owner_id      inspection_date  \
0     DA0001121     PR0000411   FA0002433       NaN  2005-03-25 00:00:00   
1     DA0001122     PR0000411   FA0002433       NaN  2005-11-07 00:00:00   
2     DA0001123     PR0000411   FA0002433       NaN  2005-11-18 00:00:00   
3     DA0001124     PR0000411   FA0002433       NaN  2006-09-25 00:00:00   
4     DA0001125     PR0000411   FA0002433       NaN  2007-07-25 00:00:00   

       inspection_time employee_id  inspection_type_id  inspection_demerits  \
0  2005-03-25 10:20:00   EE7000535                 916                  4.0   
1  2005-11-07 10:40:00   EE7000664                 916                 40.0   
2  2005-11-18 11:40:00   EE7000352                 916                  0.0   
3  2006-09-25 10:25:00   EE7000735                 916                 10.0   
4  2007-07-25 13:50:00   EE0000001                 916                  3.0   

  inspection_grade inspection_grade_new  permit_status inspection_result  \
0                A                    A            NaN         Compliant   
1                C                    C            NaN       C Downgrade   
2                A                    A            NaN         Compliant   
3                A                    A            NaN         Compliant   
4                A                    A            NaN         Compliant   

                                          violations       record_updated  \
0                                                113  2013-02-21 22:26:12   
1  12,14,21,22,25,26,27,28,30,31,36,37,64,111,112...  2013-02-21 22:26:12   
2                                                NaN  2013-02-21 22:26:12   
3                                               4,64  2013-02-21 22:26:12   
4                                           31,35,37  2013-02-21 22:26:12   

   Unnamed: 15  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN

In [22]:
restaurant_inspections = restaurant_inspections.drop('Unnamed: 15', axis = 1)
print('Shape of restaurant inspections: ', restaurant_inspections.shape)


Shape of restaurant inspections:  (265178, 15)


In [23]:
restaurant_inspections.head(20)

serial_number permit_number facility_id  owner_id      inspection_date  \
0      DA0001121     PR0000411   FA0002433       NaN  2005-03-25 00:00:00   
1      DA0001122     PR0000411   FA0002433       NaN  2005-11-07 00:00:00   
2      DA0001123     PR0000411   FA0002433       NaN  2005-11-18 00:00:00   
3      DA0001124     PR0000411   FA0002433       NaN  2006-09-25 00:00:00   
4      DA0001125     PR0000411   FA0002433       NaN  2007-07-25 00:00:00   
5      DA0001128     PR0000411   FA0002433       NaN  2008-01-14 00:00:00   
6      DA0001129     PR0000411   FA0002433       NaN  2008-10-07 00:00:00   
7      DA0001130     PR0000411   FA0002433       NaN  2009-08-25 00:00:00   
8      DA0001131     PR0000411   FA0002433       NaN  2010-03-30 00:00:00   
9      DA0001132     PR0000411   FA0002433       NaN  2010-04-13 00:00:00   
10     DA0001133     PR0000411   FA0002433       NaN  2011-03-10 00:00:00   
11     DA0001164     PR0000412   FA0002433       NaN  2005-03-25 00:00:00   
12     DA0001165     PR0000412   FA0002433       NaN  2005-11-07 00:00:00   
13     DA0001167     PR0000412   FA0002433       NaN  2005-11-16 00:00:00   
14     DA0001168     PR0000412   FA0002433       NaN  2006-09-20 00:00:00   
15     DA0001169     PR0000412   FA0002433       NaN  2007-07-25 00:00:00   
16     DA0001170     PR0000412   FA0002433       NaN  2008-01-14 00:00:00   
17     DA0001171     PR0000412   FA0002433       NaN  2008-10-07 00:00:00   
18     DA0001172     PR0000412   FA0002433       NaN  2009-08-25 00:00:00   
19     DA0001173     PR0000412   FA0002433       NaN  2010-03-30 00:00:00   

        inspection_time employee_id  inspection_type_id  inspection_demerits  \
0   2005-03-25 10:20:00   EE7000535                 916                  4.0   
1   2005-11-07 10:40:00   EE7000664                 916                 40.0   
2   2005-11-18 11:40:00   EE7000352                 916                  0.0   
3   2006-09-25 10:25:00   EE7000735                 916                 10.0   
4   2007-07-25 13:50:00   EE0000001                 916                  3.0   
5   2008-01-14 11:15:00   EE7000832                 916                 10.0   
6   2008-10-07 12:10:00   EE7000832                 916                  5.0   
7   2009-08-25 11:25:00   EE7000832                 916                  6.0   
8   2010-03-30 11:30:00   EE7000392                 916                 18.0   
9   2010-04-13 11:15:00   EE7000392                 916                  1.0   
10  2011-03-10 10:40:00   EE7000392                 916                  9.0   
11  2005-03-25 10:35:00   EE7000535                 916                  4.0   
12  2005-11-07 11:55:00   EE7000664                 916                 26.0   
13  2005-11-16 15:00:00   EE7000664                 916                 10.0   
14  2006-09-20 14:20:00   EE7000735                 916                  9.0   
15  2007-07-25 13:30:00   EE0000001                 916                  4.0   
16  2008-01-14 12:10:00   EE7000832                 916                  3.0   
17  2008-10-07 13:00:00   EE7000832                 916                  2.0   
18  2009-08-25 12:00:00   EE7000832                 916                  6.0   
19  2010-03-30 12:45:00   EE7000392                 916                  3.0   

   inspection_grade inspection_grade_new  permit_status inspection_result  \
0                 A                    A            NaN         Compliant   
1                 C                    C            NaN       C Downgrade   
2                 A                    A            NaN         Compliant   
3                 A                    A            NaN         Compliant   
4                 A                    A            NaN         Compliant   
5                 A                    A            NaN         Compliant   
6                 A                    A            NaN         Compliant   
7                 A                    A            NaN         Compliant   
8          

In [24]:
restaurant_inspections.columns

Index(['serial_number', 'permit_number', 'facility_id', 'owner_id',
       'inspection_date', 'inspection_time', 'employee_id',
       'inspection_type_id', 'inspection_demerits', 'inspection_grade',
       'inspection_grade_new', 'permit_status', 'inspection_result',
       'violations', 'record_updated'],
      dtype='object')

## The only type of inspections done are the following
#### Inspection type id: 916:  Routine inspection
####  Inspection type id: 914 : Reinspection
####  Inspection type id: 919 : Survey
#### Interestingly enough, there were no inspections from complaints

In [25]:
# Checking the type of id type of the inspections
restaurant_inspections.inspection_type_id.value_counts()

916    245754
914     19364
919        60
Name: inspection_type_id, dtype: int64

## Joining the data

In [26]:
# We want to left join NV_ businesses and reviews because each business can have multiple reviews
combinedReview = business_LV.merge(review, how= 'left', left_on='business_id', right_on='business_id')

In [61]:
business_LV.shape

(29361, 14)

In [62]:
review.shape

(6685902, 9)

In [27]:
# (2320492, 22)
combinedReview.shape

(2030063, 22)

In [38]:
combinedReview.head(5)

address  \
0  4545 E Tropicana Rd Ste 8, Tropicana   
1  4545 E Tropicana Rd Ste 8, Tropicana   
2  4545 E Tropicana Rd Ste 8, Tropicana   
3          1775 E Tropicana Ave, Ste 29   
4          1775 E Tropicana Ave, Ste 29   

                                          attributes             business_id  \
0  {'RestaurantsPriceRange2': '3', 'GoodForKids':...  gbQN7vr_caG_A1ugSmGhWg   
1  {'RestaurantsPriceRange2': '3', 'GoodForKids':...  gbQN7vr_caG_A1ugSmGhWg   
2  {'RestaurantsPriceRange2': '3', 'GoodForKids':...  gbQN7vr_caG_A1ugSmGhWg   
3  {'OutdoorSeating': 'False', 'BusinessAcceptsCr...  PZ-LZzSlhSe9utkQYU8pFg   
4  {'OutdoorSeating': 'False', 'BusinessAcceptsCr...  PZ-LZzSlhSe9utkQYU8pFg   

                                          categories       city  \
0  Hair Salons, Hair Stylists, Barbers, Men's Hai...  Las Vegas   
1  Hair Salons, Hair Stylists, Barbers, Men's Hai...  Las Vegas   
2  Hair Salons, Hair Stylists, Barbers, Men's Hai...  Las Vegas   
3                               Restaurants, Italian  Las Vegas   
4                               Restaurants, Italian  Las Vegas   

                                               hours  is_open   latitude  \
0  {'Monday': '10:0-19:0', 'Tuesday': '10:0-19:0'...        1  36.099872   
1  {'Monday': '10:0-19:0', 'Tuesday': '10:0-19:0'...        1  36.099872   
2  {'Monday': '10:0-19:0', 'Tuesday': '10:0-19:0'...        1  36.099872   
3                                                NaN        0  36.100016   
4                                                NaN        0  36.100016   

    longitude                        name  ... stars_x  state  cool  \
0 -115.074574                   Supercuts  ...     3.5     NV   0.0   
1 -115.074574                   Supercuts  ...     3.5     NV   0.0   
2 -115.074574                   Supercuts  ...     3.5     NV   1.0   
3 -115.128528  Carluccio's Tivoli Gardens  ...     4.0     NV   0.0   
4 -115.128528  Carluccio's Tivoli Gardens  ...     4.0     NV   0.0   

                  date  funny               review_id  stars_y  \
0  2015-02-18 19:39:53    0.0  DKMrV3KdKrnZAU1ML9ntJQ      5.0   
1  2015-07-23 17:53:29    0.0  UYdXyBJ_Ia1CwwzAuii5xQ      1.0   
2  2013-11-11 23:38:22    1.0  suAGAh-lVjcoPflSNcV4gg      5.0   
3  2011-06-29 02:55:07    0.0  klcF45wKIOpJW_BhJslOJg      5.0   
4  2010-10-06 18:20:13    0.0  Li-pQG6A7p5gbgZHTMeDSQ      4.0   

                                                text  useful  \
0  My stylist, Maria, listened to me carefully an...     0.0   
1  Checked in online and the app said to show in ...     0.0   
2  I won't go anywhere else. The girls at this st...     0.0   
3  We went there for dinner the other night, bein...     1.0   
4  i had the best Chicken Marcela ever. The spagh...     1.0   

                  user_id  
0  qlYas9HQ1JeiHBG2P-m4Tg  
1  hwSKzFeRh8rFHfnF3x9t0Q  
2  fRNCmStKTSCkJBkkSdwEZA  
3  -Yz2wIcsdJxUOFMbTgoKQA  
4  jYcf_e5p0UG0S-9gJq_tNA  

[5 rows x 22 columns]

In [29]:
combinedReview.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2030063 entries, 0 to 2030062
Data columns (total 22 columns):
address         object
attributes      object
business_id     object
categories      object
city            object
hours           object
is_open         int64
latitude        float64
longitude       float64
name            object
postal_code     object
review_count    int64
stars_x         float64
state           object
cool            float64
date            object
funny           float64
review_id       object
stars_y         float64
text            object
useful          float64
user_id         object
dtypes: float64(7), int64(2), object(13)
memory usage: 356.2+ MB


In [63]:
# Joining inspections data
combinedInspections = restaurant_inspections.merge(restaurant_establishments, how = 'inner', left_on = 'facility_id', right_on = 'facility_id' )


In [64]:
restaurant_inspections.shape

(265178, 15)

In [65]:
restaurant_establishments.shape

(26336, 21)

In [57]:
# (8166402, 35)
print(combinedInspections.shape)
display(combinedInspections.head(20))

(8166402, 35)


serial_number permit_number_x facility_id  owner_id_x      inspection_date  \
0      DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
1      DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
2      DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
3      DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
4      DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
5      DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
6      DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
7      DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
8      DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
9      DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
10     DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
11     DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
12     DA0001123       PR0000411   FA0002433         NaN  2005-11-18 00:00:00   
13     DA0001123       PR0000411   FA0002433         NaN  2005-11-18 00:00:00   
14     DA0001123       PR0000411   FA0002433         NaN  2005-11-18 00:00:00   
15     DA0001123       PR0000411   FA0002433         NaN  2005-11-18 00:00:00   
16     DA0001123       PR0000411   FA0002433         NaN  2005-11-18 00:00:00   
17     DA0001123       PR0000411   FA0002433         NaN  2005-11-18 00:00:00   
18     DA0001124       PR0000411   FA0002433         NaN  2006-09-25 00:00:00   
19     DA0001124       PR0000411   FA0002433         NaN  2006-09-25 00:00:00   

        inspection_time employee_id  inspection_type_id  inspection_demerits  \
0   2005-03-25 10:20:00   EE7000535                 916                  4.0   
1   2005-03-25 10:20:00   EE7000535                 916                  4.0   
2   2005-03-25 10:20:00   EE7000535                 916                  4.0   
3   2005-03-25 10:20:00   EE7000535                 916                  4.0   
4   2005-03-25 10:20:00   EE7000535                 916                  4.0   
5   2005-03-25 10:20:00   EE7000535                 916                  4.0   
6   2005-11-07 10:40:00   EE7000664                 916                 40.0   
7   2005-11-07 10:40:00   EE7000664                 916                 40.0   
8   2005-11-07 10:40:00   EE7000664                 916                 40.0   
9   2005-11-07 10:40:00   EE7000664                 916                 40.0   
10  2005-11-07 10:40:00   EE7000664                 916                 40.0   
11  2005-11-07 10:40:00   EE7000664                 916                 40.0   
12  2005-11-18 11:40:00   EE7000352                 916                  0.0   
13  2005-11-18 11:40:00   EE7000352                 916                  0.0   
14  2005-11-18 11:40:00   EE7000352                 916                  0.0   
15  2005-11-18 11:40:00   EE7000352                 916                  0.0   
16  2005-11-18 11:40:00   EE7000352                 916                  0.0   
17  2005-11-18 11:40:00   EE7000352                 916                  0.0   
18  2006-09-25 10:25:00   EE7000735                 916                 10.0   
19  2006-09-25 10:25:00   EE7000735                 916                 10.0   

   inspection_grade  ...    zip_code  nciaa plan_review record_status  \
0                 A  ...  89015-7464      Y         NaN           2.0   
1                 A  ...  89015-7464      Y         NaN           2.0   
2                 A  ...  89015-7464      Y         NaN           2.0   
3                 A  ...  89015-7464      Y         NaN           2.0   
4                 A  ...  89015-7464      Y         NaN           2.0   
5                 A  ...  89015-7464      Y         NaN           2.0   
6                 C  ...  89015-7464      Y         NaN           2.0   
7                 C  ...  89015-7464

In [58]:
combinedInspections.columns

Index(['serial_number', 'permit_number_x', 'facility_id', 'owner_id_x',
       'inspection_date', 'inspection_time', 'employee_id',
       'inspection_type_id', 'inspection_demerits', 'inspection_grade',
       'inspection_grade_new', 'permit_status', 'inspection_result',
       'violations', 'record_updated', 'permit_number_y', 'owner_id_y', 'PE',
       'restaurant_name', 'location_name', 'address', 'latitude', 'longitude',
       'city_id', 'city_name', 'zip_code', 'nciaa', 'plan_review',
       'record_status', 'current_grade', 'current_demerits', 'date_current',
       'previous_grade', 'date_previous', 'search_text'],
      dtype='object')

In [59]:
combinedInspections_names = combinedInspections.loc[:, ['facility_id', 'restaurant_name']]
combinedInspections_names.head(20)

facility_id                             restaurant_name
0    FA0002433         Black Mountain Clubhouse Restaurant
1    FA0002433                Black Mountain Clubhouse Bar
2    FA0002433       Black Mountain Clubhouse Banquet Room
3    FA0002433  Black Mountain Country Club - Golf Cart #1
4    FA0002433  Black Mountain Country Club - Golf Cart #2
5    FA0002433  Black Mountain Golf & Country Club Storage
6    FA0002433         Black Mountain Clubhouse Restaurant
7    FA0002433                Black Mountain Clubhouse Bar
8    FA0002433       Black Mountain Clubhouse Banquet Room
9    FA0002433  Black Mountain Country Club - Golf Cart #1
10   FA0002433  Black Mountain Country Club - Golf Cart #2
11   FA0002433  Black Mountain Golf & Country Club Storage
12   FA0002433         Black Mountain Clubhouse Restaurant
13   FA0002433                Black Mountain Clubhouse Bar
14   FA0002433       Black Mountain Clubhouse Banquet Room
15   FA0002433  Black Mountain Country Club - Golf Cart #1
16   FA0002433  Black Mountain Country Club - Golf Cart #2
17   FA0002433  Black Mountain Golf & Country Club Storage
18   FA0002433         Black Mountain Clubhouse Restaurant
19   FA0002433                Black Mountain Clubhouse Bar

In [60]:
combinedInspections_names.shape

(8166402, 2)

In [39]:
testing = combinedInspections.head(10)

In [40]:
testing = testing.drop_duplicates()
testing.shape

(10, 35)

In [43]:
display(testing)

serial_number permit_number_x facility_id  owner_id_x      inspection_date  \
0     DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
1     DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
2     DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
3     DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
4     DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
5     DA0001121       PR0000411   FA0002433         NaN  2005-03-25 00:00:00   
6     DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
7     DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
8     DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   
9     DA0001122       PR0000411   FA0002433         NaN  2005-11-07 00:00:00   

       inspection_time employee_id  inspection_type_id  inspection_demerits  \
0  2005-03-25 10:20:00   EE7000535                 916                  4.0   
1  2005-03-25 10:20:00   EE7000535                 916                  4.0   
2  2005-03-25 10:20:00   EE7000535                 916                  4.0   
3  2005-03-25 10:20:00   EE7000535                 916                  4.0   
4  2005-03-25 10:20:00   EE7000535                 916                  4.0   
5  2005-03-25 10:20:00   EE7000535                 916                  4.0   
6  2005-11-07 10:40:00   EE7000664                 916                 40.0   
7  2005-11-07 10:40:00   EE7000664                 916                 40.0   
8  2005-11-07 10:40:00   EE7000664                 916                 40.0   
9  2005-11-07 10:40:00   EE7000664                 916                 40.0   

  inspection_grade  ...    zip_code  nciaa plan_review record_status  \
0                A  ...  89015-7464      Y         NaN           2.0   
1                A  ...  89015-7464      Y         NaN           2.0   
2                A  ...  89015-7464      Y         NaN           2.0   
3                A  ...  89015-7464      Y         NaN           2.0   
4                A  ...  89015-7464      Y         NaN           2.0   
5                A  ...  89015-7464      Y         NaN           2.0   
6                C  ...  89015-7464      Y         NaN           2.0   
7                C  ...  89015-7464      Y         NaN           2.0   
8                C  ...  89015-7464      Y         NaN           2.0   
9                C  ...  89015-7464      Y         NaN           2.0   

  current_grade current_demerits         date_current  previous_grade  \
0             A              9.0  2018-06-06 00:00:00               A   
1             A              9.0  2018-06-06 00:00:00               A   
2             A              0.0  2017-11-11 00:00:00               A   
3             A              0.0  2016-10-05 00:00:00               P   
4             A              0.0  2016-10-05 00:00:00               P   
5             A              0.0  2017-09-20 00:00:00               A   
6             A              9.0  2018-06-06 00:00:00               A   
7             A              9.0  2018-06-06 00:00:00               A   
8             A              0.0  2017-11-11 00:00:00               A   
9             A              0.0  2016-10-05 00:00:00               P   

         date_previous                                        search_text  
0  2018-06-06 00:00:00  Black Mountain Clubhouse Restaurant Black Moun...  
1  2018-06-06 00:00:00  Black Mountain Clubhouse Bar Black Mountain Cl...  
2  2017-11-11 00:00:00  Black Mountain Clubhouse Banquet Room Black Mo...  
3  2015-10-29 00:00:00  Black Mountain Country Club - Golf Cart #1 Bla...  
4  2015-10-29 00:00:00  Black Mountain Country Club - Golf Cart #2 Bla...  
5  2017-09-20 00:00:00  Black Mountain Golf & Country Club Storage Bla...  
6  2018-06-06 00:00:00  Black Mountain Clubhouse Restaurant Black Moun...  
7  2018-06-06 00:00:00  Black Mountain Clubhou

In [42]:
testing.columns

Index(['serial_number', 'permit_number_x', 'facility_id', 'owner_id_x',
       'inspection_date', 'inspection_time', 'employee_id',
       'inspection_type_id', 'inspection_demerits', 'inspection_grade',
       'inspection_grade_new', 'permit_status', 'inspection_result',
       'violations', 'record_updated', 'permit_number_y', 'owner_id_y', 'PE',
       'restaurant_name', 'location_name', 'address', 'latitude', 'longitude',
       'city_id', 'city_name', 'zip_code', 'nciaa', 'plan_review',
       'record_status', 'current_grade', 'current_demerits', 'date_current',
       'previous_grade', 'date_previous', 'search_text'],
      dtype='object')